### DISCUSSION TITLE
Adguard Home behind Zoraxy , anyone ?

### DISCUSSION THREAD

**Discussion by palijn** [tobychui/zoraxy#272] [Q&A] [answered]
Hello,
I'm trying to access Adguard Home's GUI behind Zoraxy, without success. It seems Zoraxy cannot connect to the GUI, and I don't know why.
Zoraxy runs in a docker container, and Adguard runs in another container on the same host, in host network mode with a different IP address from the host.  ( host: 192.168.0.x ; Adguard 192.168.0.y )
Any ideas or success stories welcome !
Thanks

**Comment by PassiveLemon** [comment]
Would you mind posting your docker run/compose?

**Comment by palijn** [comment, marked as answer]
Ha! Found out what I needed to do. Adguard home in a docker container may require the use of a macvlan interface (to assign a fixed IP). And host to container communication is not possible through this interface.
Zoraxy did help me find this out because in its logs it wrote "no route to host" .

For reference, here are the steps I followed to fix this :
- add another macvlan interface
- add a route to the Adguardhome container's IP through the new macvlan
that's all folks !

On debian 12, these are the manual commands, assuming AdguardHome is running on 192.168.0.3 . : 
`ip link add dhs link enp2s0 type macvlan mode bridge`  (dhs is how I named the new interface, enp2s0 is the physical interface below) followed by `ip link set dhs up` and adding the route `ip route add 192.168.0.3/32 dev dhs`

Using NetworkManager and making it persisting across reboots is a one-liner : 
`nmcli conn add type macvlan dev enp2s0 mode bridge ifname dhs tap no ipv4.method disabled +ipv4.routes "192.168.0.3/32"`.
